modules/tracing/mule-tracer-internal-impl/src/main/java/org/mule/runtime/tracer/impl/context/EventSpanContext.java

```diff
@@ -69,6 +70,9 @@ private InternalSpan resolveParentAsInternalSpan() {
   @Override
   public void setSpan(InternalSpan span, Assertion assertion) throws AssertionFailedException {
     assertion.assertOnSpan(currentSpan);
+    if (currentSpan.getName().equals(FLOW_DISPATCHING_HANDLING_SPAN_NAME)) {
```

Comment:
This is something that will be evaluated every time we add a span to the context. In order to avoid this (plus having to know about the dispatching span name) we could invert this and add a "Dependant Span"  to the current span (that in this case will be the dispatching span)? We can get rid of the listeners also. But this is the place where we clearly see that we are coupling things a bit.